### PR TITLE
UCS/DATASTRUCT: check kh op value with defined enum

### PR DESCRIPTION
**Pull request by changchengx**
## What
use enumerated ```ucs_kh_put_t``` value to check kh operation return value.

### CODE REVIEW

**Comment by gleon99** [comment]
Need to change also inside the macros/funcs, e.g. `SCOPE khint_t kh_put_##name`

**Comment by yosefe** [comment]
> Need to change also inside the macros/funcs, e.g. `SCOPE khint_t kh_put_##name`

IMO better not change khash.h too much, in case we want to upgrade it in the future

File: src/tools/profile/read_profile.c [hunk 1]
```diff
@@ -444,7 +444,7 @@ static void show_profile_data_log(profile_data_t *data, options_t *opts,
     const char *action;
     char buf[256];
     khash_t(request_ids) reqids;
-    int hash_extra_status;
+    ucs_kh_put_t hash_extra_status;
```
[the remark this answers is not in the document]
Comment (changchengx) [reply src/tools/profile/read_profile.c:447]: @yosefe Got it. Should we unify the type of ```hash_extra_status``` in UCX code repository?
The reason that I submit this PR is that I got one CR comment to use ```ucs_kh_put_t``` type to define the related variables, so I think it's better to unify use same type to define all the related variables.

[the remark this answers is not in the document]
Comment (yosefe) [reply src/tools/profile/read_profile.c:447]: I don't think we should do it.
where is the comment?

[the remark this answers is not in the document]
Comment (changchengx) [reply src/tools/profile/read_profile.c:447]: @yosefe 
1. If so, I think we can close this PR.
2. https://github.com/openucx/ucx/pull/8956#discussion_r1176268367 (The PR is still under review, it hasn't enter into the stage to be reviewed by you until now)

[the remark this answers is not in the document]
Comment (yosefe) [reply src/tools/profile/read_profile.c:447]: > 1. If so, I think we can close this PR.

We can still fix the places that use numeric constants to use UCS_KH_PUT_xx enum

**Review by yosefe — APPROVED** [review]

**Pull request merged**